PowerEditor/src/NppIO.cpp

```diff
@@ -1598,26 +1598,54 @@ bool Notepad_plus::fileSaveSpecific(const generic_string& fileNameToSave)
 	}
 }
 
-bool Notepad_plus::fileSaveAll()
+bool Notepad_plus::fileSaveAllConfirm()
 {
-	if (viewVisible(MAIN_VIEW))
+	bool confirmed = false;
+
+	if (NppParameters::getInstance().getNppGUI()._saveAllConfirm)
 	{
-		for (size_t i = 0; i < _mainDocTab.nbItem(); ++i)
+		int answer = _nativeLangSpeaker.messageBox("SaveAllConfirm",
+			_pPublicInterface->getHSelf(),
+			TEXT("Are you sure you want to save all documents??"),
+			TEXT("Are you sure?"),
+			MB_YESNO);

```

Comment:
Use MB_YESNOCANCEL instead of MB_YESNO.
And add one sentence after "Are you sure you want to save all documents?" : 
`Choose "Cancel" if you want to dis-activate this dialog. You can always activate this option later in Preferences dialog (MISC section).`